replace_windows.go

```diff
@@ -0,0 +1,34 @@
+package cacheddownloader
+
+import (
+       "syscall"
+       "unsafe"
+)
+
+func Replace(src, dst string) error {
+	kernel32, err := syscall.LoadLibrary("kernel32.dll")
+	if err != nil {
+		return err
+	}
+	defer syscall.FreeLibrary(kernel32)
+	moveFileExUnicode, err := syscall.GetProcAddress(kernel32, "MoveFileExW")
+	if err != nil {
+		return err
+	}
+
+	srcString := syscall.StringToUTF16Ptr(src)

```

Comment:
According to https://golang.org/src/syscall/syscall_windows.go, `StringToUTF16Ptr` is deprecated in favor of `UTF16PtrFromString` (which returns an error instead of panicking on certain error cases).